tee-worker/omni-executor/executor-core/src/sync_checkpoint_repository.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020-2024 Trust Computing GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+use parity_scale_codec::{Decode, Encode};
+use std::fmt::Debug;
+use std::fs;
+use std::fs::File;
+use std::io::{ErrorKind, Write};
+
+/// Represents the point in chain. It can be a whole block or a more precise unit, for example
+/// in case of substrate chain it is BLOCK_NUM::EVENT_NUM
+pub trait Checkpoint {
+	// determines whether checkpoint is a whole block or not
+	fn just_block_num(&self) -> bool;
```

Comment:
I find `just_block_num` a bit confusing :D 

Does it make sense to have a `CheckPointType` type?

Reply:
I was thinking about it, but since it's trait we only assume there is something like 'block'.

The other improvement could look like this:
* remove `just_block_num`
* after restart always start syncing from `get_block_num() - 1` but skip already processed events